### DISCUSSION TITLE
Fully local with TeslaBleHttpProxy & Wall Connector API

### DISCUSSION THREAD

**Discussion by svwhisper** [wimaha/TeslaBleHttpProxy#16] [General]
This BLE package is great.  Combined with the Wall Connector API (see here: https://community.home-assistant.io/t/tesla-wall-connector-gen-3-restful/311670 ), I have now weaned myself off the cloud.  The Wall Connector API is good for detecting if the car is plugged in and can be charged.  That way, I don't wake the car using BLE only to find the cable is not connected.  I can also find how much power is going to the car, which of course isn't available via BLE.

A bit of Node Red to coordinate with my off-grid solar+battery system and I'm set.  Thanks again.

BTW, to the package author: You should have a "donate" link somewhere in this repo for those of us who have gotten value from your efforts.

**Comment by JasonFaulks** [comment]
Apologies for my late-to-the-party inquiry. @svwhisper it looks like you're using HomeAssistant with this proxy - do you have your Tesla Custom Integration configured to communicate directly with this BleHttpProxy or via something else? Either way, are there any tricks? I am pretty ignorant of the details but am keen to get comms going without the Tesla Fleet API drama if possible. Or have I misread your post?

**Comment by svwhisper** [comment]
I use Node Red... No HA. I do have Openhab for a gui, but all logic is in
Node Red. I'm now completely disconnected from the cloud for control of
charging. I do use Solcast, though, for solar energy forecasts.

Dave Wilson

**Reply by JasonFaulks** [reply]
Ah right, thanks. Didn't know Node Red was a thing on it's own, had only ever heard of it in conjunction with HA. At least I know you can use Node Red with the proxy - was that hard to configure?

**Comment by svwhisper** [comment]
I would say it's trivial, if you know Node Red or even just javascript.
Interfacing to the proxy is the easy part. Determining what "smart"
charging means to you is more interesting.

I'm completely satisfied with how it all works.

Dave Wilson

**Comment by sovathnahim** [comment]
How the program can communication to the car ? 
1. Do you mean , we need to connect wifi to raspi with we installed "TeslaBleHttpProxy"?
2. I still do not understand, the instruction to do it ? after input the Tesla Vin , we need to scanned the new Key card or just the known current tesla Keycard, basically tesla need to confirm by paired key card every time we set up andy key card , phone or remote.
3. Actually , I am successfully installed it but not yet testing, So please kindly advice , thanks
